examples/text_to_image/train_text_to_image.py

```diff
@@ -62,6 +63,90 @@
 }
 
 
+def make_image_grid(imgs, rows, cols):
+    assert len(imgs) == rows * cols
+
+    w, h = imgs[0].size
+    grid = Image.new("RGB", size=(cols * w, rows * h))
+
+    for i, img in enumerate(imgs):
+        grid.paste(img, box=(i % cols * w, i // cols * h))
+    return grid
+
+
+def save_model_card(
+    args,
+    repo_id: str,
+    images=None,
+    repo_folder=None,
+):
+    img_str = ""
+    if images is not None:
+        image_grid = make_image_grid(images, 1, len(args.validation_prompts))
+        image_grid.save(os.path.join(repo_folder, "val_imgs_grid.png"))
+        img_str += "![val_imgs_grid](./val_imgs_grid.png)\n"
+
+    yaml = f"""
```

Comment:
FYI `huggingface_hub` has nice utilities for model card creation (both content and metadata) and templates you could leverage here.

Reply:
See https://huggingface.co/docs/huggingface_hub/guides/model-cards

Reply:
Thanks for suggesting! 

But I feel like the current way of creating the content in the model card is a bit more explicit and also flexible. 

I could leverage something like https://huggingface.co/docs/huggingface_hub/guides/model-cards#from-a-jinja-template, but I would not here because:

* It doesn't reduce the code complexity significantly. 
* Now, users will have to learn about the model card related details of `huggingface_hub`, which I'd like to avoid because that's not the objective here. 

Cc: @pcuenca

Reply:
Thanks for the ping @osanseviero. I would also advocate to use `huggingface_hub`'s modelcards for a few reasons:
1. You need to be careful about encoding issues in the README (especially `\n` vs `\r\n` on Windows). It's quite specific and annoying but we did a few iterations in `hfh` to handle that correctly. The goal being to avoid big diffs if someone else updates the model card afterwards.
2. Using modelcards + a separate jinja template makes it really easy for non-developers to review the model card template without looking into the code. This can prove useful if someone from the ethics team (for example) wants to open a PR to complete the model card template without digging to the exact code (i.e. separate code and templates to separate usage). 
3. While it doesn't reduce much complexity, it doesn't add much either. I don't think users have to understand ModelCards internal details to use it correctly.

Reply:
That being said, I don't think the `Path('custom_template.md').write_text(template_text)` line [from the docs](https://huggingface.co/docs/huggingface_hub/guides/model-cards#from-a-jinja-template) should be reused. I think it would be best to provide a jinja template alongside the training script and only have:

```py
card_data = ModelCardData(language='en', license='mit', library_name='keras')
card = ModelCard.from_template(card_data, template_path='custom_template.md', author='nateraw')
card.save('README.md')
```

in the training script

Reply:
(thinking out loud here but) actually reusing the [default modelcard template](https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/templates/modelcard_template.md) is also an option. It simplifies the example training script on your side (no need for a custom template) as the cost of a more verbose model card -with a lot of empty fields at first-. We can see this as a way to encourage users to document better their models (which is the goal of all of this process).

Reply:
I am fine having a more verbose model card but having to provide a Jinja template separately is something I am not comfortable doing for our examples. 

> Using modelcards + a separate jinja template makes it really easy for non-developers to review the model card template without looking into the code. This can prove useful if someone from the ethics team (for example) wants to open a PR to complete the model card template without digging to the exact code (i.e. separate code and templates to separate usage).

I think they would need to open up the PR editing the README file anyway whose content is straightforward IMO. 

Probably, the best is to reuse [default modelcard template](https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/templates/modelcard_template.md) as you mentioned. Happy to accept a PR to see the changes.